mattermost-plugin/webapp/src/components/createBoardFromTemplate.tsx

```diff
@@ -73,7 +73,7 @@ const CreateBoardFromTemplate = (props: Props) => {
 
         let boardsAndBlocks = undefined
 
-        if (selectedBoardTemplateId === EMPTY_BOARD) {
+        if (templateIdRef.current === EMPTY_BOARD) {
```

Comment:
after the refactoring to this pluggable the addBoardToChannel function is sent to the webapp as an action to take after the successful channel creation, then the state value of the component is no longer valid, we needed to sent the reference instead so the actual value is correctly captured.